### PR TITLE
fix: [CDS-53499]: add tooltipid to Policy Enforcement/Execution Target

### PR DESCRIPTION
**Pull request by namanharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx [hunk 1]
```diff
@@ -192,18 +193,29 @@ export default function OptionalConfiguration(props: {
...
-            }}
-          >
-            <Radio value={'targethost'} label={'Specify Target Host'} />
-            <Radio value={'delegate'} label={'On Delegate'} />
-          </RadioGroup>
+          <Layout.Vertical>
+            <Layout.Horizontal className={css.executionTargetText}>
+              <Text
+                font={{ variation: FontVariation.FORM_LABEL }}
+                style={{ fontSize: '13px' }}
```
Comment (namanharness) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:200]: tried to add fontsize CSS to Text component but it is not working and font size needs to be 13px

https://github.com/harness/harness-core-ui/assets/110456765/f03e61c7-ea84-49b6-aaab-8ba3c68ea64f

Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:200]: Class is attached to div, for font-size to work, you can target the h1 element or pass 
font={{ variation: FontVariation.FORM_LABEL , size: FontVariation.FORM_INPUT_TEXT  }}

Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:200]: className is present on outer div, so you can need to target h1 element for font-size property to work.

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx [hunk 2]
```diff
@@ -192,18 +193,29 @@ export default function OptionalConfiguration(props: {
...
+                tooltipProps={{ dataTooltipId: 'executionTargetTooltip' }}
+              >
+                {getString('pipeline.executionTarget') + ' ' + getString('common.optionalLabel')}
+              </Text>
+            </Layout.Horizontal>
```
Comment (PratyushGar) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:205]:
```suggestion
            <Layout.Horizontal margin={{ bottom: 'medium' }}>
              <Text
                font={{ variation: FontVariation.FORM_INPUT_TEXT, weight: 'semi-bold' }}
                color={Color.GREY_600}
                tooltipProps={{ dataTooltipId: 'executionTargetTooltip' }}
              >
                {`${getString('pipeline.executionTarget')} ${getString('common.optionalLabel')}`}
              </Text>
            </Layout.Horizontal>
```

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx [hunk 3]
```diff
@@ -192,18 +193,29 @@ export default function OptionalConfiguration(props: {
...
+              data-tooltip-id="executionTargetTooltip"
+              selectedValue={formValues.spec.onDelegate}
+              disabled={readonly}
+              inline={true}
+              onChange={e => {
+                formik.setFieldValue('spec.onDelegate', e.currentTarget.value)
+              }}
+            >
+              <Radio value={'targethost'} label={'Specify Target Host'} />
+              <Radio value={'delegate'} label={'On Delegate'} />
```
Comment (PratyushGar) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:216]: Though it was already present, can we move these labels to yaml

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx (hunk 2 above)
Comment (namanharness) [reply src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:205]: will this handle fontSize as well?

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx (hunk 1 above)
Comment (namanharness) [reply src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:200]: I want to update my knowledge, shouldn't adding class to outer div add css to all its child?

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx (hunk 3 above)
Comment (namanharness) [reply src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:216]: yes, will update

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**